pkg/controller/certificates/issuing/internal/secret.go

```diff
@@ -150,36 +152,46 @@ func (s *SecretsManager) setValues(crt *cmapi.Certificate, secret *corev1.Secret
 		secret.Data[cmmeta.TLSCAKey] = data.CA
 	}
 
+	if secret.Annotations == nil {
+		secret.Annotations = make(map[string]string)
+	}
+
+	if secret.Labels == nil {
+		secret.Labels = make(map[string]string)
+	}
+
+	if crt.Spec.SecretTemplate != nil {
```

Comment:
I changed the order: first, we apply the SecretTemplate; afterwards, we apply the cert-manager annotations.
This prevents us from overwriting cert-manager annotations using the SecretTemplate.

Reply:
I'm sure that this was the original behaviour when @jonathansp originally created the feature. See
 * https://github.com/cert-manager/cert-manager/pull/3828#pullrequestreview-625178334

I wonder when and how this bug got introduced? 
Which versions are affected? Do we need to fix this bug in a separate PR and backport to the affected versions?
Is there a unit test for this? If not, please add one.

Reply:
This was changed in https://github.com/cert-manager/cert-manager/pull/4746